controllers/event/common.go

```diff
@@ -0,0 +1,136 @@
+	Port                string      `json:"port,omitempty"`
+}
+
+type BusConfig struct {
+	BusComponentName string `json:"busComponentName"`
+	BusTopic         string `json:"busTopic"`
+}
+
+type TriggerEnvConfig struct {
+	BusComponentName string               `json:"busComponentName"`
+	BusTopic         string               `json:"busTopic,omitempty"`
+	Subscribers      []*SubscriberConfigs `json:"subscribers,omitempty"`
+	Port             string               `json:"port,omitempty"`
+}
+
+type SubscriberConfigs struct {
+	SinkComponentName           string `json:"sinkComponentName,omitempty"`
+	DeadLetterSinkComponentName string `json:"deadLetterSinkComponentName,omitempty"`
+	TopicName                   string `json:"topicName,omitempty"`
+	DeadLetterTopicName         string `json:"deadLetterTopicName,omitempty"`
+}
+
+func (e *SourceEnvConfig) EncodeEnvConfig() (string, error) {
+	return encodeEnvConfig(e)
+}
+
+func (e *TriggerEnvConfig) EncodeEnvConfig() (string, error) {
+	return encodeEnvConfig(e)
+}
+
+func encodeEnvConfig(config interface{}) (string, error) {
+	envConfigBytes, err := json.Marshal(config)
+	if err != nil {
+		return "", err
+	}
+	envConfigEncode := base64.StdEncoding.EncodeToString(envConfigBytes)
+	return envConfigEncode, nil
+}
+
+func (e *SourceEnvConfig) DecodeEnvConfig(encodedConfig string) (*SourceEnvConfig, error) {
+	var config *SourceEnvConfig
+	envConifgSpec, err := decodeEnvConfig(encodedConfig)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(envConifgSpec, &config); err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+func (e *TriggerEnvConfig) DecodeEnvConfig(encodedConfig string) (*TriggerEnvConfig, error) {
+	var config *TriggerEnvConfig
+	envConifgSpec, err := decodeEnvConfig(encodedConfig)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(envConifgSpec, &config); err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+func decodeEnvConfig(encodedConfig string) ([]byte, error) {
+	if len(encodedConfig) > 0 {
+		envConifgSpec, err := base64.StdEncoding.DecodeString(encodedConfig)
+		if err != nil {
+			return nil, err
+		}
+		return envConifgSpec, nil
+	}
+	return nil, errors.New("string length is zero")
+}
+
+func CreateOrUpdateDaprComponent(client client.Client, schema *runtime.Scheme, log logr.Logger, connector *connector.Connector, object v1.Object) error {
+	ctx := context.Background()
+	// Check if component already exists
+	var component componentsv1alpha1.Component
+	if err := client.Get(ctx, types.NamespacedName{Namespace: connector.Component.Namespace, Name: connector.Component.Name}, &component); util.IsNotFound(err) {
+		log.Info("Need create dapr component", "namespace", object.GetNamespace(), "name", object.GetName())
+	}
+
+	// TODO: Determine if the dapr component in the cluster is the same as the one in the request
+	component = connector.Component
+
+	if err := client.Delete(ctx, &component); util.IgnoreNotFound(err) != nil {
```

Comment:
https://github.com/kubesphere/notification-manager/blob/master/pkg/controllers/notificationmanager_controller.go#L95

or use the hash method like https://github.com/OpenFunction/OpenFunction/blob/main/controllers/function_controller.go#L117

Maybe it's better to use similar logic like above instead of delete and then create?